core/contracts/TokenizedDerivative.sol

```diff
@@ -868,6 +879,21 @@ library TokenizedDerivativeUtils {
         }
     }
 
+    function _computeOracleFinalFee(TDS.Storage storage s, int shortBalance)
+        internal
+        view
+        returns (uint actualFeeAmount)
+    {
+        StoreInterface store = StoreInterface(_getStoreAddress(s));
+        FixedPoint.Unsigned memory expectedFee = store.computeFinalFee(
+            address(s.externalAddresses.marginCurrency)
+        );
+        uint expectedFeeAmount = expectedFee.value;
+        // Ensure the fee returned can actually be paid by the short margin account.
+        uint shortBalanceUint = _safeUintCast(shortBalance);
+        actualFeeAmount = (shortBalanceUint < expectedFeeAmount) ? shortBalanceUint : expectedFeeAmount;
```

Comment:
This brings up an interesting question around how we enforce our rules - if contracts are allowed to make the requests despite the fact that they don't have enough money to pay the final fee, that may open us up to Oracle DoS attacks from 0 value contracts. I don't think this is something we need to figure out in this PR, but we need to come up with a framework for how to deal with small contracts that cannot afford the final fee.

Reply:
Yeah that's a good point. Reverting here also seemed like a bad idea, because then it's possible that a very large balance in the long account gets stuck (for example).